tools/pd-ctl/pdctl/command/global.go

```diff
@@ -174,23 +174,29 @@ func postJSON(cmd *cobra.Command, prefix string, input map[string]interface{}) {
 		return
 	}
 
-	tryURLs(cmd, func(endpoint string) error {
+	endpoints := getEndpoints(cmd)
+	err = tryURLs(cmd, endpoints, func(endpoint string) error {
+		var msg []byte
+		var r *http.Response
 		url := endpoint + "/" + prefix
-		r, err := dialClient.Post(url, "application/json", bytes.NewBuffer(data))
+		r, err = dialClient.Post(url, "application/json", bytes.NewBuffer(data))
 		if err != nil {
 			return err
 		}
 		defer r.Body.Close()
 		if r.StatusCode != http.StatusOK {
-			if err := printResponseError(r); err != nil {
-				cmd.Println(err)
+			msg, err = ioutil.ReadAll(r.Body)
+			if err != nil {
+				return err
 			}
-			return nil
+			return errors.Errorf("[%d] %s", r.StatusCode, msg)
```

Comment:
I think we should only try next endpoint on network error. If the request was returned with a status code is not OK, there may be a problem of client, not server.

Reply:
But the error should be returned here anyway.

Reply:
But the error should be returned here anyway.